test/e2e/model/dataprovider.go

```diff
@@ -20,5 +38,42 @@ func NewTestDataProvider(keyTestPrefix string, project AProject, r *AtlasKeyType
 	for i := range initDeploymentConfigs {
 		data.Resources.Deployments = append(data.Resources.Deployments, LoadUserDeploymentConfig(data.ConfPaths[i]))
 	}
+	data.Context = context.Background()
+	k8sClient, err := kubecli.CreateNewClient()
+	if err != nil {
+		panic(fmt.Sprintf("failed to create k8s client: %v", err))
+	}
```

Comment:
I'd use `Expect` here instead of `panic`. Also, this `CreateNewClient` can be updated to use `Expect` inside as well